src/variable.jl

```diff
@@ -345,85 +345,68 @@ end
 nomissing(a::AbstractArray,value) = a
 export nomissing
 
-
-function Base.getindex(v::Variable,indexes::Int...)
+# This method needs to be duplicated instead of using an Union. Otherwise a DiskArrays fallback is called instead which impacts performances 
+# (see https://github.com/Alexander-Barth/NCDatasets.jl/pull/205#issuecomment-1589575041)
+function readblock!(v::Variable, aout, indexes::TI...) where TI <: Union{AbstractUnitRange,StepRange}
```

Comment:
```suggestion
function readblock!(v::Variable, aout, indexes::Union{<:AbstractUnitRange,<:StepRange}...)
```

This dispatch was not correct so we were using the fallback for `AbstractVector`.

I get roughly the same performance as `master` after this change.